docs/api.md

```diff
@@ -1914,6 +1920,26 @@ await page.waitFor(selector => !!document.querySelector(selector), {}, selector)
 
 Shortcut for [page.mainFrame().waitFor(selectorOrFunctionOrTimeout[, options[, ...args]])](#framewaitforselectororfunctionortimeout-options-args).
 
+#### page.waitForFileChooser([options])
```

Comment:
waitForNextFileChoser

If there is a currently active file chooser, I'd expect it to be returned by `waitForFileChooser`.

Reply:
At least add

**NOTE** This must be called BEFORE the file chooser is launched. It will not return a currently active file chooser.

Reply:
And if in the future we got the ability to return the currently active file picker, would we do it with this same API?

Reply:
> And if in the future we got the ability to return the currently active file picker, would we do it with this same API?

I think technically there's no difference between getting the currently active file picker and the one that's opening - so I guess yes.

Reply:
> At least add

> NOTE This must be called BEFORE the file chooser is launched. It will not return a currently active file chooser.

Done.